src/query/server/query.go

```diff
@@ -172,6 +184,109 @@ func Run(runOpts RunOptions) {
 
 	xconfig.WarnOnDeprecation(cfg, logger)
 
+	if cfg.MultiProcess.Enabled {
+		multiProcessInstance := os.Getenv(multiProcessInstanceEnvVar)
+		if multiProcessInstance == "" {
+			logger = logger.With(zap.String("processID", "parent"))
+			count := cfg.MultiProcess.Count
+			if count == 0 {
```

Comment:
nit: `count < 1` to be safe? Can log the value if it's ignored due to being non zero